packages/pancake-uikit/src/components/BottomDrawer/BottomDrawer.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useRef } from "react";
+import { createPortal } from "react-dom";
+import useOnClickOutside from "../../hooks/useOnClickOutside";
+import useDelayedUnmount from "../../hooks/useDelayedUnmount";
+import { DrawerContainer, StyledOverlay } from "./styles";
+import { CloseIcon } from "../Svg";
+import { Box } from "../Box";
+import { IconButton } from "../Button";
+import { useMatchBreakpoints } from "../..";
+
+const portalRoot = document.getElementById("portal-root");
+
+interface BottomDrawerProps {
+  content: React.ReactNode;
+  isOpen: boolean;
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const BottomDrawer: React.FC<BottomDrawerProps> = ({ content, isOpen, setIsOpen }) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const shouldRender = useDelayedUnmount(isOpen, 350);
+  const { isMobile } = useMatchBreakpoints();
+
+  useOnClickOutside(ref, () => setIsOpen(false));
+
+  if (!shouldRender || !isMobile) {
+    document.body.style.overflow = "visible";
+    return null;
+  }
+
+  // Prevent scrolling of the page when drawer is shown
+  document.body.style.overflow = "hidden";
```

Comment:
better wrap in useEffect?

Reply:
hm.... could be. But it would effectively do same thing but with more lines of code (+ overhead of calling useEffect).
Logic here is that if it's hidden (`return null` above) we set it overflow to normal behaviour. But if its not hidden (i.e. above if statement didn't fire) we continue to here and hide overflow.

Might be that resetting the style for every BottomDrawer rerender is not performant... but I'm not sure that useEffect overhead is any better to be honest 😅